### ISSUE TITLE
Library Binary Pattern

### ISSUE THREAD

**Issue by pinkforest** [Skallwar/suckit#124]
@CohenArthur @Skallwar Are you okay if I change to lib-bin pattern from binary only now?

Essentially I would split all the library logic to library and binary that requires different things e.g. for shell related libraries.

This is a common pattern and will make organising code, benches, testing and dependency management easier between two sets of needs among other things.

Also common pattern is that library will never panic whilst binary may panic but idea is to propagate errors to binary.

**Comment by CohenArthur** [comment]
Yup, this is a really good idea. This is something that we had thought about earlier, but never got around to doing. I'd love to see it implemented.

**Comment by pinkforest** [comment]
@CohenArthur @Skallwar initial lib-bin split is there, 

On top need to push:
- move out panics & error propagation (quick_error will be good for this)
- tests to use library not binary execution
- dependencies by optional features (maybe, not that important I guess for now)

**Comment by Skallwar** [comment]
Add a checkbox list to this issue to make it a tracking issue of the patch-set

**Comment by Skallwar** [comment]
I would also like to see the argument parsing moved to the bin

**Comment by Skallwar** [comment]
Hi @pinkforest, I hope you are doing well. I was wondering if you are not planing to followup on this or if you are just taking a break?
